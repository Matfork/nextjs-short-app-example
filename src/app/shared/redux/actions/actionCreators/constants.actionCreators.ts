export const AC_MENU = {
  MENU_CLEAR: 'menu/clear',
  MENU_REQUEST: 'menu/request',
  MENU_ERROR: 'menu/error',
  MENU_SUCCESS: 'menu/success'
};
