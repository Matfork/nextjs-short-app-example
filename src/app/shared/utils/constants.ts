export const _C = {
  IS_BROWSER: (process as any).browser
};
