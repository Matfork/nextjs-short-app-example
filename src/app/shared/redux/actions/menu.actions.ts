import { AC_MENU } from './actionCreators/constants.actionCreators';

// Login Methods
export const onMenuRequest = (
  data: any
  // setErrors: any,
  // setSubmitting: any
) => ({
  type: AC_MENU.MENU_REQUEST,
  data
  // setErrors,
  // setSubmitting
});
export const onMenuSuccess = (data: any) => ({
  type: AC_MENU.MENU_SUCCESS,
  data
});
export const onMenuError = (data: any) => ({
  type: AC_MENU.MENU_ERROR,
  data
});
export const onMenuClear = () => ({
  type: AC_MENU.MENU_CLEAR
});
