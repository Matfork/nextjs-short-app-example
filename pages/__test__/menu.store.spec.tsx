import configureStore from 'redux-mock-store';
import * as menuActions from '../../src/app/shared/redux/actions/menu.actions';
import * as menuReducers from '../../src/app/shared/redux/reducers/menu.reducer';
import * as menuActionCreators from '../../src/app/shared/redux/actions/actionCreators/constants.actionCreators';

const mockStore = configureStore();
const store = mockStore();

describe('menu_actions', () => {
  beforeEach(() => {
    // Runs before each test in the suite
    store.clearActions();
  });

  test('Dispatches the correct action and payload', () => {
    // const expectedActions = [
    //   {
    //     data: '',
    //     type: menuActionCreators.AC_MENU.MENU_REQUEST
    //   }
    // ];

    store.dispatch(menuActions.onMenuRequest(''));
    //expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });
});

describe('menu_reducer', () => {
  describe('INITIAL_STATE', () => {
    test('is correct', () => {
      const action = { type: '' };
      const initialState = {
        loading: false,
        data: null,
        error: null
      };

      expect(menuReducers.default(undefined, action)).toEqual(initialState);
    });
  });

  describe('MENU_REQUEST', () => {
    test('returns correct data', () => {
      const action = {
        type: menuActionCreators.AC_MENU.MENU_REQUEST,
        data: {}
      };
      const expectedState = { data: null, error: null, loading: true };
      expect(menuReducers.default(undefined, action)).toEqual(expectedState);
    });
  });

  describe('MENU_SUCCESS', () => {
    test('returns correct data', () => {
      const menu = [
        {
          value: 1,
          label: 'user_id'
        },
        {
          value: 2,
          label: 'item_id'
        },
        {
          value: 3,
          label: 'user_email'
        },
        {
          value: 4,
          label: 'item_views'
        },
        {
          value: 5,
          label: 'item_favorites'
        }
      ];

      const action = {
        type: menuActionCreators.AC_MENU.MENU_SUCCESS,
        data: {
          data: menu
        }
      };

      // const expectedState = {
      //   data: menu,
      //   error: null,
      //   loading: false
      // };
      //expect(menuReducers.default(undefined, action)).toEqual(expectedState);
      expect(menuReducers.default(undefined, action)).toMatchSnapshot();
    });
  });
});
