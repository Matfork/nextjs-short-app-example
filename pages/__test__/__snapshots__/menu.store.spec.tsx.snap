// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu_actions Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "data": "",
    "type": "menu/request",
  },
]
`;

exports[`menu_reducer MENU_SUCCESS returns correct data 1`] = `
Object {
  "data": Array [
    Object {
      "label": "user_id",
      "value": 1,
    },
    Object {
      "label": "item_id",
      "value": 2,
    },
    Object {
      "label": "user_email",
      "value": 3,
    },
    Object {
      "label": "item_views",
      "value": 4,
    },
    Object {
      "label": "item_favorites",
      "value": 5,
    },
  ],
  "error": null,
  "loading": false,
}
`;
